src/Tools/dotnet-user-jwts/src/Helpers/SigningKeysHandler.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Configuration.UserSecrets;
+using Microsoft.Extensions.Configuration;
+using System.Text.Json.Nodes;
+using System.Text.Json;
+using System.Linq;
+using System.Globalization;
+
+namespace Microsoft.AspNetCore.Authentication.JwtBearer.Tools;
+
+internal static class SigningKeysHandler
+{
+    public static byte[] GetSigningKeyMaterial(string userSecretsId, string scheme, string issuer)
+    {
+        var projectConfiguration = new ConfigurationBuilder()
+            .AddUserSecrets(userSecretsId)
+            .Build();
+
+        var signingKey = projectConfiguration
+            .GetSection(GetSigningKeyPropertyName(scheme))
+            .Get<SigningKey[]>()
+            ?.SingleOrDefault(key => key.Issuer == issuer);
+        var signingKeyLength = signingKey?.Length ?? DevJwtsDefaults.SigningKeyLength;
+
+        var keyMaterial = new byte[signingKeyLength];
+        if (!string.IsNullOrEmpty(signingKey?.Value)
+            && Convert.TryFromBase64String(signingKey.Value, keyMaterial, out var bytesWritten)
+            && bytesWritten == signingKeyLength)
+        {
+            return keyMaterial;
+        }
+
+        return null;
+    }
+
+    public static byte[] GetOrCreateSigningKeyMaterial(string userSecretsId, string scheme, string issuer) =>
+        GetSigningKeyMaterial(userSecretsId, scheme, issuer) ?? CreateSigningKeyMaterial(userSecretsId, scheme, issuer);
+ 
+
+    public static byte[] CreateSigningKeyMaterial(string userSecretsId, string scheme, string issuer, int signingKeyLength = 32, bool reset = false)
+    {
+        // Create signing material and save to user secrets
+        var newKeyMaterial = System.Security.Cryptography.RandomNumberGenerator.GetBytes(signingKeyLength);
+        var secretsFilePath = PathHelper.GetSecretsPathFromSecretsId(userSecretsId);
+        Directory.CreateDirectory(Path.GetDirectoryName(secretsFilePath));
+
+        JsonObject secrets = null;
+        if (File.Exists(secretsFilePath))
+        {
+            using var secretsFileStream = new FileStream(secretsFilePath, FileMode.Open, FileAccess.Read);
+            if (secretsFileStream.Length > 0)
+            {
+                secrets = JsonSerializer.Deserialize<JsonObject>(secretsFileStream);
+            }
+        }
+
+        secrets ??= new JsonObject();
+        var signkingKeysPropertyName = GetSigningKeyPropertyName(scheme);
+        var shortId = Guid.NewGuid().ToString("N").Substring(0, 8);
+        var key = new SigningKey(shortId, issuer, Convert.ToBase64String(newKeyMaterial), signingKeyLength);
+
+        if (secrets.ContainsKey(signkingKeysPropertyName))
+        {
+            var signingKeys = secrets[signkingKeysPropertyName].AsArray();
+            if (reset)
```

Comment:
Ok, so if multiple keys for the same issuer is allowed, does reset need to delete all keys from the same issuer now?

Reply:
So, to start, I'd only added support for multiple keys for the same issuer in the runtime, but not in the user-jwts tool. My rationale for this was that we would keep the surface area of the tool simpler than what the runtime could support. Users would still be able to support the multiple-keys-per-issuer if they set it up in config themselves, but the tool wouldn't support it.

I realize the incongruency is kinda weird though so maybe we should stick to one-key-per-issuer to keep things simple throughout the board (I want to avoid modifying too much of the existing handling in an RC1 PR).

Reply:
> I realize the incongruency is kinda weird though

Yeah, it's strange 😌, unless we have a scenario for multiple keys in the Runtime I don't feel like we should add it yet until both areas are updated.

Reply:
I've definitely seen examples of multiple keys from the same issuer being configured so leaving it in the runtime might be useful for now with the prospect of adding support to the tool later on.